vyper/codegen/jumptable_utils.py

```diff
@@ -0,0 +1,195 @@
+# helper module which implements jumptable for function selection
+import math
+from dataclasses import dataclass
+
+from vyper.utils import method_id_int
+
+
+@dataclass
+class Signature:
+    method_id: int
+    payable: bool
+
+
+# bucket for dense function
+@dataclass
+class Bucket:
+    bucket_id: int
+    magic: int
+    method_ids: list[int]
+
+    @property
+    def image(self):
+        return _image_of([s for s in self.method_ids], self.magic)
+
+    @property
+    # return method ids, sorted by by their image
+    def method_ids_image_order(self):
+        return [x[1] for x in sorted(zip(self.image, self.method_ids))]
+
+    @property
+    def bucket_size(self):
+        return len(self.method_ids)
+
+
+BITS_MAGIC = 24  # a constant which produced good results, see _bench()
```

Comment:
Do you mean `_bench_dense` and `_bench_sparse` or is there somewhere else a function I missed? I'm also wondering whether the implementation of library modules will impact this magic somehow...

Reply:
ah, i meant `_bench_dense`. i don't think it is affected by modules, since the selector table is calculated as the last step in contract bytecode construction. in any case, we will cross that bridge when we get there :)

Reply:
fixed in 12e4b39bc